Create BonoBocadillo.cs   "#22"

carlos122211 opened the pull request:
Closes #22
Contexto
- CU: Bonos de Bocadillo.
- Primera pasada sin relaciones (solo atributos propios y validaciones).

Alcance de esta PR
- Clase Models/BonoBocadillo.cs con DataAnnotations:
  - BonoId (PK no identity, Range >= 1)
  - Nombre (Required, StringLength 2..120)
  - NBocadillos (Range >= 1)
  - CantidadDisponible (Range >= 0)
  - PVP (decimal(10,2), Range >= 0)
- Validacion cruzada: CantidadDisponible <= NBocadillos.
- Registro en DbContext (sin DbSet o con DbSet, segun el proyecto).
- Actualizado diagrama .cd (si aplica).

Fuera de alcance (se hara en otra task)
- Propiedades de navegacion y claves foraneas.
- Reglas de negocio dependientes de otras entidades.

Pruebas / Verificacion rapida
- Compila y pasa los checks del repo.
- Crear instancia valida e invalida (Nombre vacio, PVP < 0, etc.) y verificar DataAnnotations.
- Diff del PR sin archivos bajo Migrations/.

Riesgos y notas
- No subir migraciones (politica del repo).
- Mantener un unico ApplicationDbContext en /Data para evitar conflictos.

carlos122211 committed:
Create BonoBocadillo.cs   "#22"

Contexto
- CU: Bonos de Bocadillo.
- Primera pasada sin relaciones (solo atributos propios y validaciones).

Alcance de esta PR
- Clase Models/BonoBocadillo.cs con DataAnnotations:
  - BonoId (PK no identity, Range >= 1)
  - Nombre (Required, StringLength 2..120)
  - NBocadillos (Range >= 1)
  - CantidadDisponible (Range >= 0)
  - PVP (decimal(10,2), Range >= 0)
- Validacion cruzada: CantidadDisponible <= NBocadillos.
- Registro en DbContext (sin DbSet o con DbSet, segun el proyecto).
- Actualizado diagrama .cd (si aplica).

Fuera de alcance (se hara en otra task)
- Propiedades de navegacion y claves foraneas.
- Reglas de negocio dependientes de otras entidades.

Pruebas / Verificacion rapida
- Compila y pasa los checks del repo.
- Crear instancia valida e invalida (Nombre vacio, PVP < 0, etc.) y verificar DataAnnotations.
- Diff del PR sin archivos bajo Migrations/.

Riesgos y notas
- No subir migraciones (politica del repo).
- Mantener un unico ApplicationDbContext en /Data para evitar conflictos.

RamonUCLM commented on src/AppForSEII2526.API/Models/BonoBocadillo.cs:
```diff
@@ -0,0 +1,48 @@
+namespace AppForSEII2526.Models  // <-- cambia al namespace real
+{
+   
+     Entidad BonoBocadillo (primera pasada: sin relaciones).
```
Comenta esta línea para que no de error al compilar.

carlos122211 replied:
ya lo he solucionado

RamonUCLM reviewed (changes requested):
Comenta la línea 9 porque no va a compilar el proyecto ni se podrán crear las migraciones.

carlos122211 committed:
Update BonoBocadillo.cs

ya he solucionado el error

carlos122211 committed:
Update BonoBocadillo.cs

he añadido lo correspondiente para las relaciones

carlos122211 committed:
Update BonoBocadillo.cs

me he equivocado de branch y no he añadido las relaciones a la bran de las relaciones entre los atributos

RamonUCLM reviewed (approved):
La clase está bien, queda aprobado el PR.
